ui/dashboards/src/components/Variables/VariableEditor.tsx

```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
-              Apply
-            </Button>
-            <Button
-              variant="outlined"
-              onClick={() => {
-                props.onCancel();
-              }}
-            >
-              Cancel
-            </Button>
-          </Stack>
-          <Typography variant="h3" mb={2}>
-            Variable List
-          </Typography>
-          <Stack spacing={2}>
-            {!validation.isValid &&
-              validation.errors.map((error) => (
-                <Alert severity="error" key={error}>
-                  {error}
-                </Alert>
-              ))}
-            <TableContainer component={Paper}>
-              <Table sx={{ minWidth: 650 }} aria-label="simple table">
-                <TableHead>
-                  <TableRow>
-                    <TableCell>Visibility</TableCell>
-                    <TableCell>Variable Name</TableCell>
-                    <TableCell>Variable Type</TableCell>
-                    <TableCell align="right">Action</TableCell>
-                  </TableRow>
-                </TableHead>
-                <TableBody>
-                  {variableDefinitions.map((v, idx) => (
-                    <TableRow key={v.spec.name}>
-                      <TableCell component="th" scope="row">
-                        <Switch
-                          checked={v.spec.display?.hidden !== true}
-                          onChange={(e) => {
-                            toggleVariableVisibility(idx, e.target.checked);
-                          }}
-                        />
-                      </TableCell>
-                      <TableCell component="th" scope="row" sx={{ fontWeight: 'bold' }}>
-                        {v.spec.name}
-                      </TableCell>
-                      <TableCell>{v.kind}</TableCell>
-                      <TableCell align="right">
-                        <IconButton onClick={() => changeVariableOrder(idx, 'up')} disabled={idx === 0}>
-                          <ArrowUp />
-                        </IconButton>
-                        <IconButton
-                          onClick={() => changeVariableOrder(idx, 'down')}
-                          disabled={idx === variableDefinitions.length - 1}
-                        >
-                          <ArrowDown />
-                        </IconButton>
-
-                        <IconButton onClick={() => setVariableEditIdx(idx)}>
-                          <PencilIcon />
-                        </IconButton>
-                        <IconButton onClick={() => removeVariable(idx)}>
-                          <TrashIcon />
-                        </IconButton>
-                      </TableCell>
-                    </TableRow>
-                  ))}
-                </TableBody>
-              </Table>
-            </TableContainer>
-            <Box display="flex">
-              <Button onClick={addVariable} variant="contained">
-                Add New Variable
+          <Box
```

Comment:
Once we figure out the right abstraction, I think some common "helper" components would be nice to keep this consistent across features - similar to what @juliepagano did. I think @eunicorn may have started doing some of this for the other too

Reply:
yeah we have a `Drawer` component, maybe we should include the header there as well

Reply:
Oh I like that - I was thinking about where a shared component for the header could possibly go

@saminzadeh which helper component did Julie make?